### PR TITLE
Update CMIP6 CMOR tables to version 01.00.31 (6.8.31)

### LINKED ISSUE #364: sftlf error in ERA-Interim data
The `sftlf` variable is not correctly cmorized in `ERA-Interim`.
When running `recipe_check_obs.yml` the following is returned:
```
"""
Traceback (most recent call last):
  File "/mnt/lustre01/pf/b/b309057/SOFTWARE/miniconda3/envs/esmvaltool/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_task.py", line 678, in _run_task
    output_files = task.run()
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_task.py", line 230, in run
    self.output_files = self._run(input_files)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/preprocessor/__init__.py", line 403, in _run
    product.apply(step, self.debug)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/preprocessor/__init__.py", line 271, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/preprocessor/__init__.py", line 213, in preprocess
    result.append(_run_preproc_function(function, items, settings))
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/preprocessor/__init__.py", line 199, in _run_preproc_function
    return function(items, **kwargs)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/cmor/fix.py", line 145, in fix_metadata
    cube = checker(cube).check_metadata()
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/cmor/check.py", line 107, in check_metadata
    self.report_errors()
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/cmor/check.py", line 126, in report_errors
    raise CMORCheckError(msg)
esmvalcore.cmor.check.CMORCheckError: There were errors in variable sftlf:
typeland: does not exist
 sftlf: does not match coordinate rank
in cube:
land_area_fraction / (%)            (-- : 1; latitude: 241; longitude: 480)
     Dimension coordinates:
          latitude                      -            x               -
          longitude                     -            -               x
     Attributes:
          Conventions: CF-1.5
          comment: 'Contains modified Copernicus Climate Change Service Information 2019'
          host: mlogin102
          mip: fx
          modeling_realm: reanaly
          project_id: OBS6
          reference: Dee et al., Q. J. Roy. Meteor. Soc., doi:10.1002/qj.828, 2011.
          source: www.ecmwf.int/en/forecasts/datasets/reanalysis-datasets/era-interim
          source_file: /mnt/lustre02/work/bd0854/DATA/ESMValTool2/OBS/Tier3/ERA-Interim/OBS6_...
          tier: 3
          title: ERA-Interim data reformatted for ESMValTool v2.0.0b1
          user: b309057
          version: 1
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_main.py", line 228, in run
    conf = main(args)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_main.py", line 156, in main
    process_recipe(recipe_file=recipe, config_user=cfg)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_main.py", line 206, in process_recipe
    recipe.run()
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_recipe.py", line 1288, in run
    max_parallel_tasks=self._cfg['max_parallel_tasks'])
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_task.py", line 599, in run_tasks
    _run_tasks_parallel(tasks, max_parallel_tasks)
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_task.py", line 645, in _run_tasks_parallel
    _copy_results(task, running[task])
  File "/mnt/lustre01/pf/b/b309057/ESMValTool/core/esmvalcore/_task.py", line 666, in _copy_results
    task.output_files, updated_products = future.get()
  File "/mnt/lustre01/pf/b/b309057/SOFTWARE/miniconda3/envs/esmvaltool/lib/python3.7/multiprocessing/pool.py", line 657, in get
    raise self._value
esmvalcore.cmor.check.CMORCheckError: There were errors in variable sftlf:
typeland: does not exist
 sftlf: does not match coordinate rank
in cube:
land_area_fraction / (%)            (-- : 1; latitude: 241; longitude: 480)
     Dimension coordinates:
          latitude                      -            x               -
          longitude                     -            -               x
     Attributes:
          Conventions: CF-1.5
          comment: 'Contains modified Copernicus Climate Change Service Information 2019'
          host: mlogin102
          mip: fx
          modeling_realm: reanaly
          project_id: OBS6
          reference: Dee et al., Q. J. Roy. Meteor. Soc., doi:10.1002/qj.828, 2011.
          source: www.ecmwf.int/en/forecasts/datasets/reanalysis-datasets/era-interim
          source_file: /mnt/lustre02/work/bd0854/DATA/ESMValTool2/OBS/Tier3/ERA-Interim/OBS6_...
          tier: 3
          title: ERA-Interim data reformatted for ESMValTool v2.0.0b1
          user: b309057
          version: 1
```

I think the time coordinate has to be removed.

### PR DESCRIPTION
**Pull request by bouweandela**
**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Related to https://github.com/ESMValGroup/ESMValCore/issues/329, not sure if it solves that completely though.

Closes #364

### CODE REVIEW

File: esmvalcore/cmor/tables/cmip6/Tables/CMIP6_Oclim.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
         "int_missing_value": "-999", 
         "product": "model-output", 
         "approx_interval": "30.00000", 
-        "generic_levels": "olevel", 
+        "generic_levels": "olevel olevhalf",
```
Comment (valeriupredoi) [inline esmvalcore/cmor/tables/cmip6/Tables/CMIP6_Oclim.json]: can anyone check this is not in any of the issues on the pcmdi github page pls? @bouweandela @zklaus @jvegasbsc

Comment (valeriupredoi) [reply esmvalcore/cmor/tables/cmip6/Tables/CMIP6_Oclim.json]: https://github.com/PCMDI/cmor/issues/559

**Comment by valeriupredoi** [comment]
@bouweandela any reason why you pulled `6.8.31` and not the stable version `01.00.31`?

**Comment by valeriupredoi** [comment]
not so fast @mattiarighi there are two standing issues: the actual version itself (see my comment above) and the issue with `olevhalf` that is currently looked into by the PCMDI folk :beer:

**Comment by bouweandela** [comment]
> @bouweandela any reason why you pulled 6.8.31 and not the stable version 01.00.31?

6.8.31 is the latest release of the CMIP6 CMOR tables, see: https://github.com/PCMDI/cmip6-cmor-tables/releases. It contains version 01.00.31 of the CMIP6 data request.

**Comment by mattiarighi** [comment]
It would be good to have a quick go at one of the CMIP6 recipes with this new version @schluma @debe-kevin

**Comment by valeriupredoi** [comment]
Cool! cheers @bouweandela :beer: I reckon we should fix that Oclim table
since the PCMDI guys have confirmed the bug (see issue I opened on their
github page)

Dr Valeriu Predoi.
Computational scientist
NCAS-CMS
University of Reading
Department of Meteorology
Reading RG6 6BB
United Kingdom

**Comment by valeriupredoi** [comment]
@schlunma

Dr Valeriu Predoi.
Computational scientist
NCAS-CMS
University of Reading
Department of Meteorology
Reading RG6 6BB
United Kingdom

**Comment by debe-kevin** [comment]
> 
> 
> It would be good to have a quick go at one of the CMIP6 recipes with this new version @schlunma @debe-kevin

Hello @mattiarighi, 
I tested the PR it on a rather simple recipe (consecdrydays, see below txt of the recipe). 

[recipe_consecdrydays_cmip6.txt](https://github.com/ESMValGroup/ESMValCore/files/3824813/recipe_consecdrydays_cmip6.txt)

No error to report. I may run it on a more tricky recipe and let you know if I spot an error !

**Comment by mattiarighi** [comment]
If you have the cmip6 version of `recipe_perfmetrics` at hand it would be good to try it.
Thank you!

**Comment by bouweandela** [comment]
@debe-kevin Any news?

**Comment by debe-kevin** [comment]
> 
> 
> @debe-kevin Any news?

Hello Bouwe,
I ran the perfmetrics recipe on CMIP6 below. The recipe requires some fix files that I will push soon. Everything worked fine after adding the updated CMIP6 CMOR table :) !

[recipe_perfmetrics_cmip6.txt](https://github.com/ESMValGroup/ESMValCore/files/3850518/recipe_perfmetrics_cmip6.txt)

**Comment by bouweandela** [comment]
Thanks for testing! That is good news. @mattiarighi can this be merged then?

**Comment by mattiarighi** [comment]
>  The recipe requires some fix files that I will push soon. 

Are these fixes required because of the updated CMOR tables?
If this is the case we should merge the fix files first, otherwise this PR would prevent some recipes to run.

**Comment by debe-kevin** [comment]
> 
> 
> > The recipe requires some fix files that I will push soon.
> 
> Are these fixes required because of the updated CMOR tables?
> If this is the case we should merge the fix files first, otherwise this PR would prevent some recipes to run.

@mattiarighi no there are fixes for missing coordinates that I needed in order to run the recipe before including the PR commits.

**Comment by bouweandela** [comment]
Yes, I would hope this pull request will reduce the number of fixes needed, since it removes some erroneously introduced auxiliary coordinates on fx files, see e.g. #364.

**Pull request merged**